docs/sql-ref-literals.md

```diff
@@ -0,0 +1,506 @@
+{% endhighlight %}
+
+#### Parameters
+
+<dl>
+  <dt><code><em>digit</em></code></dt>
+  <dd>
+    Any numeral from 0 to 9.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>D</em></code></dt>
+  <dd>
+    Case insensitive, indicates <code>DOUBLE</code>, which is a 8-byte double-precision floating point number.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>BD</em></code></dt>
+  <dd>
+    Case insensitive, indicates <code>BIGDECIMAL</code>, which is an arbitrary-precision signed decimal number.
+  </dd>
+</dl>
+<dl>
+  <dt><code><em>default (no postfix)</em></code></dt>
+  <dd>
+    Indicate a 4-byte single-precision floating point number.
+  </dd>
+</dl>
+
+#### Examples
+
+{% highlight sql %}
+SELECT 5E2 AS col;
+  +-----+
+  |  col|
+  +-----+
+  |500.0|
+  +-----+
+
+SELECT 5D AS col;
+  +---+
+  |col|
+  +---+
+  |5.0|
+  +---+
+
+SELECT -5BD AS col;
+  +---+
+  |col|
+  +---+
+  | -5|
+  +---+
+
+SELECT 12.578e-2d AS col;
+  +-------+
+  |    col|
+  +-------+
+  |0.12578|
+  +-------+
+
+SELECT -.1234567E+2BD AS col;
+  +---------+
+  |      col|
+  +---------+
+  |-12.34567|
+  +---------+
+
+SELECT +3.e+3 AS col;
+  +------+
+  |   col|
+  +------+
+  |3000.0|
+  +------+
+
+SELECT -3.E-3D AS col;
+  +------+
+  |   col|
+  +------+
+  |-0.003|
+  +------+
+{% endhighlight %}
+
+### Datetime Literal
+
+A Datetime literal is used to specify a datetime value.
+
+#### Date Literal
+
+#### Syntax
+
+{% highlight sql %}
+DATE { 'yyyy' |
+       'yyyy-[m]m' |
+       'yyyy-[m]m-[d]d' |
+       'yyyy-[m]m-[d]d ' |
+       'yyyy-[m]m-[d]d[T]c[...]' }
+{% endhighlight %}
+
+#### Parameters
+
+<dl>
+  <dt><code><em>c</em></code></dt>
+  <dd>
+    One character from the character set.
+  </dd>
+</dl>
+
+#### Examples
+
+{% highlight sql %}
+SELECT DATE '1997' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |1997-01-01|
+  +----------+
+
+SELECT TIMESTAMP '1997-01' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |1997-01-01|
+  +----------+
+
+SELECT DATE '2011-11-11' AS col;
+  +----------+
+  |       col|
+  +----------+
+  |2011-11-11|
+  +----------+
+{% endhighlight %}
+
+#### Timestamp Literal
+
+#### Syntax
+
+{% highlight sql %}
+TIMESTAMP { 'yyyy' |
+            'yyyy-[m]m' |
+            'yyyy-[m]m-[d]d' |
+            'yyyy-[m]m-[d]d ' |
+            'yyyy-[m]m-[d]d[T][h]h[:]' |
+            'yyyy-[m]m-[d]d[T][h]h:[m]m[:]' |
+            'yyyy-[m]m-[d]d[T][h]h:[m]m:[s]s[.]' |
+            'yyyy-[m]m-[d]d[T][h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]' |
+            '[T][h]h:' |
+            '[T][h]h:[m]m[:]' |
+            '[T][h]h:[m]m:[s]s[.]' |
+            'T[h]h' |
+            '[T][h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]' }
+{% endhighlight %}
+`zone_id` should have one of the forms:
+<ul>
+  <li>Z - Zulu time zone UTC+0</li>
+  <li>+|-[h]h:[m]m</li>
+  <li>A short id, see <a href="https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS">SHORT_IDS</a></li>
```

Comment:
Referencing java doc seems not appropriate in Spark SQL doc as we created a datetime pattern as our own for the same reason. cc @cloud-fan

Reply:
I removed the link and explicitly list them all